diff --git a/ext/ffi/lib.rs b/ext/ffi/lib.rs
index de4ff3ef2..eac54018f 100644
--- a/ext/ffi/lib.rs
+++ b/ext/ffi/lib.rs
@@ -27,6 +27,7 @@ use std::path::Path;
 use std::path::PathBuf;
 use std::ptr;
 use std::rc::Rc;
+use std::os::raw::c_char;
 
 pub struct Unstable(pub bool);
 
@@ -608,7 +609,7 @@ where
   let permissions = state.borrow_mut::<FP>();
   permissions.check(None)?;
 
-  let ptr = u64::from(ptr) as *const i8;
+  let ptr = u64::from(ptr) as *const c_char;
   Ok(unsafe { CStr::from_ptr(ptr) }.to_str()?.to_string())
 }
 
