diff --git a/test_napi/src/typedarray.rs b/test_napi/src/typedarray.rs
index fed63773e..b874f2bc0 100644
--- a/test_napi/src/typedarray.rs
+++ b/test_napi/src/typedarray.rs
@@ -4,6 +4,7 @@ use crate::assert_napi_ok;
 use crate::napi_get_callback_info;
 use crate::napi_new_property;
 use core::ffi::c_void;
+use std::os::raw::c_char;
 use napi_sys::Status::napi_ok;
 use napi_sys::TypedarrayType;
 use napi_sys::ValueType::napi_number;
@@ -103,7 +104,7 @@ extern "C" fn test_multiply(
     assert_napi_ok!(napi_throw_error(
       env,
       ptr::null(),
-      "Typed array was of a type not expected by test.".as_ptr() as *const i8,
+      "Typed array was of a type not expected by test.".as_ptr() as *const c_char,
     ));
     return ptr::null_mut();
   }
